### PR TITLE
VirtualScroller: performance enhancements

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
- Removes unused props from a variety of chat components
- Prevents scroll adjustments from transcluded nodes from being render-blocking (useEffect vs useLayoutEffect)
- Renders less chatmessages at once
- Makes BigIntOrderedMap immutable, to correctly leverage React memoisation
- GraphContentWide now correctly memoizes
- Prevents refs from being reset every render

ChatMessage: remove dead props

ChatResource: remove unnecessary props

ChatWindow: remove unnecessary props

interface: remove dead pendings

virtualContext: drop useLayoutEffect

VirtualScroller: rework for less memory use, faster speeds

interface: make BigIntOrderedMap immutable

interface: remove unnecessary routing props

### CODE REVIEW

**Review by matildepark — APPROVED** [review]

**Comment by matildepark** [comment]
Appears so, yes.

**Comment by matildepark** [comment]
There is one weird thing, though, I would suggest looking at: if you use this to proxy onto dabben-larbet, and you go to Interface chat in the Tlon group, it uh, crashes the tab.

**Comment by tacryt-socryp** [comment]
Well, let's throw it onto next-js then lmao

**Review by tacryt-socryp — APPROVED** [review]
Idk the difference between useLayoutEffect and useEffect but this is good to go.

**Comment by matildepark** [comment]
Wait wait wait don't merge, crashing the tab is sort of a big deal.

**Review by matildepark — CHANGES_REQUESTED** [review]
What's going on with that case ...

File: pkg/interface/src/views/components/VirtualScroller.tsx [hunk 1]
```diff
@@ -575,8 +574,10 @@ function VirtualChild(props: VirtualChildProps) {
 
   const ref = useCallback((el: HTMLElement | null) => {
     setRef(el, props.index);
-  }, [setRef, props.index])
+  //  VirtualChild should always be keyed on the index, so the index should be
+  //  valid for the entire lifecycle of the component, hence no dependencies
+  }, []);
```
Comment (tacryt-socryp) [inline pkg/interface/src/views/components/VirtualScroller.tsx:579]: I believe this makes it update on every tick

Comment (liam-fitzgerald) [reply pkg/interface/src/views/components/VirtualScroller.tsx:579]: No, that's if the dependencies are not provided. `[]` will cause the ref callback to be constant throughout the entire lifecycle.

**Comment by matildepark** [comment]
The unread banner straight up doesn't appear in these "big unread backlog" cases, which is a regression.

**Review by matildepark — APPROVED** [review]

**Pull request merged**